tests/testthat/test-plot_with_settings_ui.R

```diff
@@ -6,39 +62,427 @@ testthat::test_that("plot_with_settings_ui returns `shiny.tag.list`", {
+  "e2e: teal.widgets::plot_with_settings: initializes with a plot and toggle, download and resize buttons",
+  {
+    skip_if_too_deep(5)
+    app_driver <- shinytest2::AppDriver$new(
+      app_driver_pws(),
+      name = "pws",
+      variant = "app_driver_pws_ui"
+    )
+    app_driver$wait_for_idle(timeout = default_idle_timeout)
+
+    # Check if there is an image.
+    testthat::expect_true(is_visible("#plot_with_settings-plot_main > img", app_driver))
+
+    # Check if there are three buttons above the table.
+    testthat::expect_true(is_visible("#plot_with_settings-downbutton-downl", app_driver))
+    testthat::expect_true(is_visible("#plot_with_settings-expand", app_driver))
+    testthat::expect_true(is_visible("#plot_with_settings-expbut", app_driver))
+
+    app_driver$stop()
+  }
+)
+
+testthat::test_that(
+  "e2e: teal.widgets::plot_with_settings: buttons have proper FA icons and two of them are dropdowns",
+  {
+    skip_if_too_deep(5)
+    app_driver <- shinytest2::AppDriver$new(
+      app_driver_pws(),
+      name = "pws",
+      variant = "app_driver_pws_ui"
+    )
+    app_driver$wait_for_idle(timeout = default_idle_timeout)
+
+    testthat::expect_equal(
+      app_driver$get_html("#plot_with_settings-downbutton-downl") %>%
+        rvest::read_html() %>%
+        rvest::html_element("button") %>%
+        rvest::html_attr("data-toggle"),
+      "dropdown"
+    )
+
+    testthat::expect_equal(
+      app_driver$get_html("#plot_with_settings-downbutton-downl") %>%
+        rvest::read_html() %>%
+        rvest::html_element("button") %>%
+        rvest::html_attr("aria-expanded"),
+      "false"
+    )
+
+    testthat::expect_equal(
+      app_driver$get_html("#plot_with_settings-downbutton-downl") %>%
+        rvest::read_html() %>%
+        rvest::html_element("i") %>%
+        rvest::html_attr("class"),
+      "fas fa-download"
+    )
+
+    testthat::expect_equal(
+      app_driver$get_html("#plot_with_settings-expand") %>%
+        rvest::read_html() %>%
+        rvest::html_element("i") %>%
+        rvest::html_attr("class"),
+      "fas fa-up-right-and-down-left-from-center"
+    )
+
+    testthat::expect_equal(
+      app_driver$get_html("#plot_with_settings-expbut") %>%
+        rvest::read_html() %>%
+        rvest::html_element("i") %>%
+        rvest::html_attr("class"),
+      "fas fa-maximize"
+    )
+
+    testthat::expect_equal(
+      app_driver$get_html("#plot_with_settings-expbut") %>%
+        rvest::read_html() %>%
+        rvest::html_element("button") %>%
+        rvest::html_attr("data-toggle"),
+      "dropdown"
+    )
+
+    testthat::expect_equal(
+      app_driver$get_html("#plot_with_settings-expbut") %>%
+        rvest::read_html() %>%
+        rvest::html_element("button") %>%
+        rvest::html_attr("aria-expanded"),
+      "false"
+    )
+
+    app_driver$stop()
+  }
+)
+
+testthat::test_that(
+  "e2e: teal.widgets::plot_with_settings: the click on the first button opens a download menu
```

Comment:
It doesn't click the first button - then css selector would look like `<button>:nth-child(1)`
```suggestion
  "e2e: teal.widgets::plot_with_settings: the click on the download button opens a download menu
```